extensions/amp-brid-player/0.1/amp-brid-player.js

```diff
@@ -47,63 +71,80 @@ class AmpBridPlayer extends AMP.BaseElement {
     return isLayoutSizeDefined(layout);
   }
 
+  getVideoIframeSrc_() {
+    if (this.videoIframeSrc_) {
+      return this.videoIframeSrc_;
+    }
+
+    let feedType = '';
+
+    if (this.element.hasAttribute('data-video')) {
+      feedType = 'video';
+    } else if (this.element.hasAttribute('data-playlist')) {
+      feedType = 'playlist';
+    }
+
+    //Create iframe
+    const src = 'https://services.brid.tv/services/iframe/' +
+        encodeURIComponent(feedType) +
+        '/' + encodeURIComponent(this.feedID_) +
+        '/' + encodeURIComponent(this.partnerID_) +
+        '/' + encodeURIComponent(this.playerID_) + '/0/1';
+
+    this.videoIframeSrc_ = assertAbsoluteHttpOrHttpsUrl(src);
+
+    return this.videoIframeSrc_;
+  }
+
   /** @override */
   buildCallback() {
+
     this.partnerID_ = user().assert(
         this.element.getAttribute('data-partner'),
         'The data-partner attribute is required for <amp-brid-player> %s',
         this.element);
 
+    this.playerID_ = user().assert(this.element.getAttribute('data-player'),
+        'The data-player attribute is required for <amp-brid-player> %s',
+        this.element);
+
     this.feedID_ = user().assert(
         (this.element.getAttribute('data-video') ||
         this.element.getAttribute('data-playlist')),
         'Either the data-video or the data-playlist ' +
         'attributes must be specified for <amp-brid-player> %s',
         this.element);
+
+    this.playerReadyPromise_ = new Promise(resolve => {
+      this.playerReadyResolver_ = resolve;
+    });
+
+    installVideoManagerForDoc(this.element);
+    videoManagerForDoc(this.element).register(this);
   }
 
   /** @override */
   layoutCallback() {
-    const playerID = user().assert(this.element.getAttribute('data-player'),
-        'The data-player attribute is required for <amp-brid-player> %s',
-        this.element);
-
-    const partnerID = user().assert(
-        this.partnerID_,
-        'The data-partner attribute is required for <amp-brid-player> %s',
-        this.element);
-
-    let feedType = '';
-
-    if (this.element.getAttribute('data-video')) {
-      feedType = 'video';
-    } else if (this.element.getAttribute('data-playlist')) {
-      feedType = 'playlist';
-    }
-
     //Create iframe
     const iframe = this.element.ownerDocument.createElement('iframe');
-    const src = 'https://services.brid.tv/services/iframe/' +
-        encodeURIComponent(feedType) +
-        '/' + encodeURIComponent(this.feedID_) +
-        '/' + encodeURIComponent(partnerID) +
-        '/' + encodeURIComponent(playerID) + '/0/1';
     iframe.setAttribute('frameborder', '0');
     iframe.setAttribute('allowfullscreen', 'true');
-    iframe.src = src;
+    iframe.src = this.getVideoIframeSrc_();
     this.applyFillContent(iframe);
     this.element.appendChild(iframe);
     this.iframe_ = iframe;
-    return this.loadPromise(iframe);
+
+    this.win.addEventListener(
```

Comment:
Please use 
```
    this.unlistenMessage_ = listen(
      this.win,
      'message',
      this. handleBridMessages_.bind(this)
    );
```

and define this.unlistenMessage_ in the contractor.

We also need a new `unlayoutCallback` override that does:

```
  /** @override */
  unlayoutCallback() {
    if (this.iframe_) {
      removeElement(this.iframe_);
      this.iframe_ = null;
    }
    if (this.unlistenMessage_) {
      this.unlistenMessage_();
    }
    this.playerState_ = PlayerStates.PAUSED;

    this.playerReadyPromise_ = new Promise(resolve => {
      this.playerReadyResolver_ = resolve;
    });
    return true;  // Call layoutCallback again.
  }
```